### PR TITLE
Implement skinned mesh animation pipeline with Lua bindings (stable base, follow-up needed for perf/render issues)

### PR DESCRIPTION
**Pull request by Gopmyc**
## Description
This PR introduces the core skinned mesh animation pipeline and Lua interaction for animated models.

Implemented:
- Added `CSkinnedMeshRenderer` runtime component for skeletal animation playback.
- Integrated automatic component attachment/synchronization when using compatible skinned models.
- Added animation runtime controls (play/pause/stop, clip selection, looping, speed, time).
- Integrated skinning data flow to rendering.
- Added Lua bindings so scripts can interact with the skinned animation component:
  - `Actor:GetSkinnedMeshRenderer()`
  - `Actor:AddSkinnedMeshRenderer()`
  - `Actor:RemoveSkinnedMeshRenderer()`
  - `SkinnedMeshRenderer` methods (playback, clip selection, speed/time, state queries).

Current state:
- The implementation is stable enough for use and collaboration.
- Previously observed memory/access issues were addressed and are no longer reproducing in current tests.
- Lua API interaction is working.

Known issues intentionally left for follow-up:
- There are still intermittent performance drops in some situations.
- Some mesh faces (notably on hands/fingers in certain assets) may fail to render.
- Root cause for missing faces is still not clearly identified (could involve import/weights/render path).
- These issues should be handled in a dedicated PR to keep this one focused on the core animation/skinning foundation.

This PR is submitted now to share the working base with the team and unblock parallel review/contributions.

## Related Issue(s)
N/A

## Review Guidance
- Load a scene with a skinned model.
- Verify `CSkinnedMeshRenderer` is present and animation plays correctly.
- Validate controls in runtime/inspector (clip switch, pause/play, speed, looping, time).
- Validate Lua access to the component and methods listed above.
- Confirm no memory/access crash occurs during repeated play/stop and scene interactions.
- Note: intermittent perf drops and missing hand faces are known follow-up items.

## Checklist
- [x] My code follows the project's code style guidelines
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the documentation accordingly
- [x] My changes don't generate new warnings or errors

## Wiki Updates Required

The PR adds Lua bindings for `CSkinnedMeshRenderer`, so the wiki must be updated with exact API signatures and behavior.

### 1) Update `Scripting-API.md`

In **Built-in components**, add:
- [SkinnedMeshRenderer](SkinnedMeshRenderer)

### 2) Update `Component.md`

In the list of usertypes inheriting from `Component`, add:
- [SkinnedMeshRenderer](SkinnedMeshRenderer)

### 3) Update `Actor.md`

Add these functions in the Actor functions table:

|Name|Input|Output|Description|
|-|-|-|-|
|GetSkinnedMeshRenderer|`N/A`&nbsp;<br>|`SkinnedMeshRenderer`|Returns the SkinnedMeshRenderer attached to this actor (if any, otherwise `nil`)|
|AddSkinnedMeshRenderer|`N/A`&nbsp;<br>|`SkinnedMeshRenderer`|Adds a SkinnedMeshRenderer component to the actor and returns it|
|RemoveSkinnedMeshRenderer|`N/A`&nbsp;<br>|`boolean`|Removes the SkinnedMeshRenderer component from the actor (returns `true` on success)|

### 4) Add new page `SkinnedMeshRenderer.md`

Use this content:

### Description
A component responsible for skeletal animation playback and skinning runtime control.

- C++ related class: `OvCore::ECS::Components::CSkinnedMeshRenderer`
- Inherits from lua usertype: [Component](Component)

### Variables
_This usertype has no variables_

### Constructors
_This usertype has no constructors_

### Functions
|Name|Input|Output|Description|
|-|-|-|-|
|NotifyModelChanged|`N/A`&nbsp;<br>||Forces internal resync after model-related changes|
|HasCompatibleModel|`N/A`&nbsp;<br>|`boolean`|Returns `true` if the current model is skinned and compatible|
|HasSkinningData|`N/A`&nbsp;<br>|`boolean`|Returns `true` if a valid skinning palette is currently available|
|SetEnabled|`boolean`:&nbsp;enabled<br>||Enables/disables the component|
|IsEnabled|`N/A`&nbsp;<br>|`boolean`|Returns whether the component is enabled|
|Play|`N/A`&nbsp;<br>||Starts/resumes animation playback|
|Pause|`N/A`&nbsp;<br>||Pauses animation playback|
|Stop|`N/A`&nbsp;<br>||Stops playback and resets time to start|
|IsPlaying|`N/A`&nbsp;<br>|`boolean`|Returns whether playback is active|
|SetLooping|`boolean`:&nbsp;looping<br>||Sets looping mode|
|IsLooping|`N/A`&nbsp;<br>|`boolean`|Returns whether looping is enabled|
|SetPlaybackSpeed|`number`:&nbsp;speed<br>||Sets animation playback speed|
|GetPlaybackSpeed|`N/A`&nbsp;<br>|`number`|Returns current playback speed|
|SetTime|`number`:&nbsp;timeSeconds<br>||Sets playback time in seconds|
|GetTime|`N/A`&nbsp;<br>|`number`|Returns playback time in seconds|
|GetAnimationCount|`N/A`&nbsp;<br>|`number`|Returns the number of available animation clips|
|GetAnimationName|`number`:&nbsp;index<br>|`string`|Returns clip name at index|
|SetAnimation|`number`:&nbsp;index<br>|`boolean`|Sets active clip by **0-based index**, returns `true` on success|
|SetAnimation|`string`:&nbsp;name<br>|`boolean`|Sets active clip by name, returns `true` on success|
|GetAnimationIndex|`N/A`&nbsp;<br>|`number`|Returns current clip index (`-1` if none)|
|GetAnimation|`N/A`&nbsp;<br>|`string`|Returns current clip name (empty string if none)|

### Operators
_This usertype has no operators_

### Notes
- `SetAnimation(index)` uses a **0-based** index.
- If no compatible skinned model is set, control functions may safely no-op or return failure.
- Typical usage is through `Actor:GetSkinnedMeshRenderer()` or `Actor:AddSkinnedMeshRenderer()`.

## Collaboration Note

Animation support is not fully finalized yet, and some areas still need polishing.  
This PR is opened intentionally to share the current implementation early, gather feedback, and collaborate on the remaining work.

At this stage, the foundation is stable and usable:
- no known crashes during current tests
- no known memory leaks in the tested scenarios
- Lua interaction is functional

Follow-up improvements (quality/performance/edge cases) will be addressed in subsequent PRs.

### CODE REVIEW

File: Resources/Engine/Shaders/Common/Buffers/SkinningSSBO.ovfxh [hunk 1]
```diff
@@ -0,0 +1,50 @@
+layout(std430, binding = 2) buffer SkinningSSBO
+{
+    mat4 ssbo_Bones[];
+};
+
+mat4 ComputeSkinningMatrix(vec4 p_boneIDs, vec4 p_boneWeights)
```
Comment (adriengivry) [inline Resources/Engine/Shaders/Common/Buffers/SkinningSSBO.ovfxh]: This should be separated from the SkinningSSBO definition. Maybe adding it to a new `Engine/Shaders/Common/Skinning.ovfxh` would make sense

File: Resources/Editor/Shaders/OutlineFallback.ovfx [hunk 2]
```diff
@@ -1,7 +1,63 @@
...
+#feature SKINNING
+
 #shader vertex
 #version 450 core
 
 layout(location = 0) in vec3 geo_Pos;
+layout(location = 5) in vec4 geo_BoneIDs;
+layout(location = 6) in vec4 geo_BoneWeights;
+
+layout(std430, binding = 2) buffer SkinningSSBO
```
Comment (adriengivry) [inline Resources/Editor/Shaders/OutlineFallback.ovfx]: Not a fan of duplicating this code. We should figure out if we could include engine shaders (SkinningSSBO.ovfhx) in editor shaders.
